src/main.ts

```diff
@@ -156,11 +156,11 @@ export async function run(): Promise<void> {
       resumable: resumable,
       predefinedAcl: predefinedAcl,
 
-      onUploadObject: (source: string, destination: string, opts: Record<string, unknown>) => {
+      onUploadObject: (source: string, destination: string, opts: UploadOptions) => {
         core.info(`Uploading ${source} to gs://${destination}`);
 
         if (core.isDebug()) {
-          const data = deepClone(opts);
+          const data = deepClone(opts) as Record<string, unknown>;
```

Comment:
Why not keep this UploadOptions. Is there a reason?

Reply:
We're adding our own custom fields which aren't part of UploadOptions below (like timestamp)